R/module_data_summary.R

```diff
@@ -0,0 +1,261 @@
+#' Data summary
```

Comment:
note: this is present in `main`

Is an error appearing in summary the expected behavior when a `FilterStateExpr` fails?

I would assume a smaller error on the summary such as "Error applying filters. See error on main panel."

<details>

<summary>Example app</summary>

```r
options(
  teal.log_level = "INFO",
  teal.show_js_log = TRUE,
  # teal.bs_theme = bslib::bs_theme(version = 5),
  shiny.bookmarkStore = "server"
)

pkgload::load_all("teal.data")
pkgload::load_all("teal.slice")
pkgload::load_all("teal")

data <- teal_data(ADSL = teal.data::rADSL)

teal::init(
  data = data,
  modules = teal::modules(
    teal::example_module(label = "A"),
    teal::example_module(label = "B")
  ),
  filter = teal::teal_slices(
    # FilterExpr
    teal_slice(
      dataname = "ADSL",
      id = "Female adults",
      expr = "SEX == 'F' & AGEA >= 18",
      title = "Female adults"
    ),
    module_specific = FALSE
  ),
  title = "Example App"
) |>
  shiny::runApp()
```

</details>

![image](https://github.com/user-attachments/assets/796a2843-1f24-489b-8482-48fed840a1f5)

### Mockup of what could be the solution

![image](https://github.com/user-attachments/assets/5deeea10-7ce9-4b9f-a747-9f1c369e72e8)

Reply:
Simple PR here: #1294

based on `crayon::strip_styles` (we could use this instead, but implies adding 1 more dependency only for this call)

Reply:
FYI: `crayon` is a superseded package. It's advised to replace it with `cli`. Now assuming we already have it in deps - maybe there is a better alternative there?

Reply:
Unfortunately, we are not using `cli` on teal directly.

I would prefer to use `cli::ansi_strip`, but my point still stands as adding 1 dependency in imports for this is overkill.

My suggestion on the PR was to copy the regular expression to clean the string and avoid this dependency.